src/pathOrderOptimizer.h

```diff
@@ -17,17 +37,15 @@ namespace cura {
 class PathOrderOptimizer
 {
 public:
-    EZSeamType type;
     Point startPoint; //!< A location near the prefered start location
-    Point z_seam_pos; //!< The position near where to create the z_seam (if \ref PathOrderOptimizer::type == 'back')
+    const ZSeamConfig* config;
```

Comment:
You made this member a pointer type so that it can be optional, while ZSeamConfig has a default constructor which looks as if it's meant exactly for that.

I think the whole charm of introducing the ZSeamConfig class is that you don't need a parameter with a default value which therefore has to be at the end of the function signature. I think that instead of working with `ZSeamConfig*` you should just work with `const ZSeamConfig&`.